xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -41,15 +43,19 @@ using namespace PVR;
 using namespace EPG;
 
 CGUIWindowPVRGuide::CGUIWindowPVRGuide(bool bRadio) :
-  CGUIWindowPVRBase(bRadio, bRadio ? WINDOW_RADIO_GUIDE : WINDOW_TV_GUIDE, "MyPVRGuide.xml")
+  CGUIWindowPVRBase(bRadio, bRadio ? WINDOW_RADIO_GUIDE : WINDOW_TV_GUIDE, "MyPVRGuide.xml"),
+  m_pRefreshTimelineItemsThread(new CPVRRefreshTimelineItemsThread(this)),
+  m_bRefreshTimelineItems(false)
 {
-  m_bUpdateRequired = false;
   m_cachedTimeline = new CFileItemList;
   m_cachedChannelGroup = CPVRChannelGroupPtr(new CPVRChannelGroup);
 }
 
 CGUIWindowPVRGuide::~CGUIWindowPVRGuide(void)
 {
+  StopRefreshTimelineItemsThread();
```

Comment:
the destructor of a window is not called when the window is closed. the extra thread would run until application exit which is not desired.